sbws/core/cleanup.py

```diff
@@ -20,12 +24,44 @@ def gen_parser(sub):
 
     :param argparse._SubParsersAction sub: what to add a sub-parser to
     '''
-    d = 'Compress data files that are no longer fresh and delete data files '\
-        'that stopped being fresh a long time ago'
+    d = 'Compress and delete results and/or v3bw files old files.' \
+        'Configuration options are read to determine which are old files'
     p = sub.add_parser('cleanup', description=d,
                        formatter_class=ArgumentDefaultsHelpFormatter)
     p.add_argument('--dry-run', action='store_true',
                    help='Don\'t actually compress or delete anything')
+    p.add_argument('--no-results', action='store_true',
+                   help='Do not clean results files')
+    p.add_argument('--no-v3bw', action='store_true',
+                   help='Do not clean v3bw files')
+
+
+def _get_files_mtime_older_than(dname, days_delta, extensions):
+    """Return files which modification time is older than days_delta
+    and which extension is one of the extensions."""
+    assert os.path.isdir(dname)
+    assert isinstance(days_delta, int)
+    assert isinstance(extensions, list)
+    for ext in extensions:
+        assert isinstance(ext, str)
+        assert ext[0] == '.'
+    # Determine oldest allowed date
+    today = datetime.utcfromtimestamp(time.time())
+    oldest_day = today - timedelta(days=days_delta)
+    for root, dirs, files in os.walk(dname):
+        for f in files:
+            fname = os.path.join(root, f)
+            _, ext = os.path.splitext(fname)
+            if ext not in extensions:
+                log.debug('Ignoring %s because it doesn\'t have extension '
+                          '%s', fname, ext)
```

Comment:
"Because it doesn't have an extension in [the list of extensions]"

Reply:
`ext` is the extension the file *does* have. While `extensions` is the list of allowed extensions. The second half of my comment above was pointing out we're logging

`Ignoring foo.bin becuase it doesn't have extension .bin`

When we should be logging

`Ignoring foo.bin because its extension is not in ['.v3bw', 'v3bw.gz']`